src/pybullet_blocks/envs/cleanup_table_env.py

```diff
@@ -0,0 +1,768 @@
+        """Depth of the interior of the bin."""
+        return self.bin_size[1] - 2 * self.bin_wall_thickness
+
+    @property
+    def interior_height(self) -> float:
+        """Height of the interior of the bin."""
+        return self.bin_size[2] - self.bin_bottom_thickness
+
+    @property
+    def interior_bottom_z_offset(self) -> float:
+        """Z offset from bin origin to interior bottom."""
+        return -self.bin_size[2] / 2 + self.bin_bottom_thickness
+
+
+@dataclass(frozen=True)
+class CleanupTableSceneDescription(BaseSceneDescription):
+    """Container for toy cleanup task hyperparameters."""
+
+    num_toys: int = 3
+
+    # Bin parameters
+    bin_position: tuple[float, float, float] = (0.95, 0.0, -0.375)
+    bin_rgba: tuple[float, float, float, float] = (0.6, 0.4, 0.2, 1.0)
+    bin_dimensions: BinDimensions = field(default_factory=BinDimensions)
+
+    # Toy parameters
+    toy_rgba: tuple[float, float, float, float] = (0.8, 0.2, 0.2, 1.0)
+    toy_half_extents: tuple[float, float, float] = (0.1, 0.1, 0.1)
+    toy_mass: float = 0.3
+    toy_friction: float = 0.9
+
+    # Objaverse configuration
+    objaverse_config: ObjaverseConfig = field(default_factory=ObjaverseConfig)
+    use_objaverse: bool = True
+
+    # Table position
+    table_pose: Pose = Pose((0.5, 0.0, -0.175))
+    table_half_extents: tuple[float, float, float] = (0.3, 0.3, 0.25)
+
+    @property
+    def toy_init_position_lower(self) -> tuple[float, float, float]:
+        """Lower bounds for toy position on table."""
+        return (
+            self.table_pose.position[0]
+            - self.table_half_extents[0]
+            + self.toy_half_extents[0],
+            self.table_pose.position[1]
+            - self.table_half_extents[1]
+            + self.toy_half_extents[1],
+            self.table_pose.position[2]
+            + self.table_half_extents[2]
+            + self.toy_half_extents[2],
+        )
+
+    @property
+    def toy_init_position_upper(self) -> tuple[float, float, float]:
+        """Upper bounds for toy position on table."""
+        return (
+            self.table_pose.position[0]
+            + self.table_half_extents[0]
+            - self.toy_half_extents[0],
+            self.table_pose.position[1]
+            + self.table_half_extents[1]
+            - self.toy_half_extents[1],
+            self.table_pose.position[2]
+            + self.table_half_extents[2]
+            + self.toy_half_extents[2],
+        )
+
+
+@dataclass(frozen=True)
+class CleanupTablePyBulletBlocksState(PyBulletBlocksState):
+    """A state in the toy cleanup environment with graph representation."""
+
+    toy_states: Collection[LetteredBlockState]
+    bin_state: BlockState
+    robot_state: RobotState
+
+    @classmethod
+    def get_node_dimension(cls) -> int:
+        """Get the dimensionality of nodes."""
+        return max(
+            RobotState.get_dimension(),
+            LetteredBlockState.get_dimension(),
```

Comment:
I'm all for reusing code but I think `LetteredBlock` and `Block` are now misleading. should we change their names to something more general? (as a side note, we should probably change the name of this entire repository at some point)